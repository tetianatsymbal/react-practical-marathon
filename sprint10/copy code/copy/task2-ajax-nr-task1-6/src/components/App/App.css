@import "./styleNull.css";
@import "../Categories/Categories.css";
@import "../Characters/Characters.css";