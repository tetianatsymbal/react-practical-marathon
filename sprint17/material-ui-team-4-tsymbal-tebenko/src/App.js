import { ThemeProvider } from "@mui/material";
import Banner from "./components/Banner/Banner";
import Content from "./components/Content/Content.jsx";
import Footer from "./components/Footer/Footer.jsx";
import Header from "./components/header/Header.jsx";
import theme from "./styles";

export default function App() {
   return <ThemeProvider theme={theme}>
            <Header />
            <Banner/>
            <Content/>
            <Footer/>
          </ThemeProvider>
}
