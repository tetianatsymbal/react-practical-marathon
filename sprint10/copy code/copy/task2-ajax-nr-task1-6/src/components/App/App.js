import '../../../src/components/App/App.css';
import Categories from '../Categories/Categories'
// import InfoPage from '../infoPage/InfoPage'

function App() {
  return (
    <div className="App">
      <Categories />
    </div>
  );
}

export default App;
