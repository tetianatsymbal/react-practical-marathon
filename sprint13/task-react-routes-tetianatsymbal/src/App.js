import React, { useState } from 'react';
import { Routes, Route, useParams, useSearchParams, Link, useNavigate } from 'react-router-dom';
import './App.css';


const AppWrapper1 = () => {
  const { param } = useParams();
  return <div>Subtask1, parameter: {param}</div>
};

const AppWrapper2 = () => {
  const [searchParams] = useSearchParams ();
  let queryParams = searchParams.toString().split("&").join(", ");
  // console.log(queryParams);
  return <div>Subtask2, query parameters: {queryParams}</div>
};

const AppWrapper3 = () => {
  return <div>Subtask3, protected information</div>
};

const AppWrapper4 = () => {
  return <div>Subtask4, navigated programmatically</div>
};


export default function App() {

  const [inputNum, setInputNum] = useState("");

  const navigated = useNavigate();
  
  return (
    <div className="App">
      <h1>React Marathon</h1>
      <h2>The topic 'Routes'</h2>
      <Link to={ inputNum % 2 !== 0 ? '/subtask3' : "/" } >Show protected information if</Link>
      <span>&nbsp;</span>
      <input size="5" onChange={(e)=>setInputNum(e.target.value)} value={inputNum}></input> is odd
      <div className="mainClass">
        Go to the component programmatically, by checking the box:{" "}
        <input type="checkbox" checked={false} onChange={()=>navigated('/subtask4')} ></input>
      </div>
      <Routes>
        <Route path='/subtask1/:param' element={<AppWrapper1/>} />
        <Route path='/subtask2/*' element={<AppWrapper2/>} />
        <Route path='/subtask3' element={<AppWrapper3/>} />
        <Route path='/subtask4' element={<AppWrapper4/>} />
      </Routes>
    </div>
  );
}
