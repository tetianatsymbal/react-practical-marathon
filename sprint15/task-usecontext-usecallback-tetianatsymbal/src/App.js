import React, {useState} from "react";
import './App.css';
import Contacts from './contacts';
import Logo from "./Logo";

export const ContactContext = React.createContext();

function App() {
  
  const [contacts, setContacts] = useState([
    { index: 0, channelOption: '', details: '' }
  ]);
 
  const value = {
    contacts,
    setContacts
  };
  return (
    <ContactContext.Provider value={value}>
      <div className="grid-container">
        <div>
          <Contacts contacts={contacts} setContacts={setContacts} />
        </div>
        <div>       
          <Logo />
        </div>
      </div>
    </ContactContext.Provider>
  );
}

export default App;
