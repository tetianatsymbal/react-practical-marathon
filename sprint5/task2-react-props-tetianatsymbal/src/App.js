import React from "react";
import "./App.css";
import First from "./First";

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      list: [
        { id: 1, text: "Animals" },
        { id: 2, text: "Anime" },
        { id: 3, text: "Anti-Malware" },
        { id: 4, text: "Art Design" },
        { id: 5, text: "Books" },
        { id: 6, text: "Business" },
        { id: 7, text: "Calendar" },
        { id: 8, text: "Cloud Storage" },
        { id: 9, text: "File Sharing" },
        { id: 10, text: "Animals" },
        { id: 11, text: "Continuous Integration" },
        { id: 12, text: "Cryptocurrency" },
      ],
    };
  }
  render() {
    const list = this.state.list.map((el) => el.text.toLowerCase());
    return (
      <div>
        Some data:
        <ul>
          <First listArr={list} />
        </ul>
      </div>
    );
  }
}
