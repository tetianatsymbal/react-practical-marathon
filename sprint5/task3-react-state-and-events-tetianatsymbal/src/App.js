import React from "react";
import "./App.css";

export default class App extends React.Component {
  state = {
    appData: "React Marathon",
  };

  onChangeClick = () => {
    this.setState((prev) => ({
      appData: prev.appData.toLowerCase(),
    }));
  };

  render() {
    return <div onClick={this.onChangeClick}> {this.state.appData} </div>;
  }
}
