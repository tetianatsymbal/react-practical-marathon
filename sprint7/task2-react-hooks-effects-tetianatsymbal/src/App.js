import { useEffect, useState } from "react";
import "./App.css";

export default function App() {
  const [appData, setAppData] = useState("");

  useEffect(() => {
    const value = localStorage.getItem("appData");
    setAppData(value);
  }, [appData]);

  return (
    <div>
      React Marathon, appData:<input size="5" defaultValue={appData}></input>
    </div>
  );
}
