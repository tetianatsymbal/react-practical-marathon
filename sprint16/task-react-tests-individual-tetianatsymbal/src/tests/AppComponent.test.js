import { render } from "@testing-library/react";
import App from "../components/App";

const response = { data: { id: 1, name: "tania" } };
jest.mock("../services/DataService", () => () => Promise.resolve(response));

describe("App", () => {
  it("renders correctly", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test("App should render div element className: App", async () => {
    expect(App().props.className).toEqual("App");
  });

  test("App number of props should be: 3", async () => {
    expect(App().props.children.length).toEqual(3);
  });
});


// TODO: Your test need to be here instead of fake tests