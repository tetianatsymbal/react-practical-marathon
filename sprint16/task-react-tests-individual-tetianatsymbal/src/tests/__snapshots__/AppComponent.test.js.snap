// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Hello World!
    </h1>
    <div>
      <h3>
        GitHub User Info
      </h3>
      <ul />
    </div>
    <div>
      <h3>
        GitHub User Info
      </h3>
      <ul />
    </div>
  </div>
</div>
`;
